db/migrate/20250320121115_antwerpbuildingtext.rb

```diff
@@ -0,0 +1,65 @@
+class Antwerpbuildingtext < ActiveRecord::Migration[5.0]
+  def self.up
+    directory    = Rails.root.join('db/migrate/20250320121115_antwerpbuildingtext')
+    data_path    = directory.join('data.csv')
+    commits_path = directory.join('commits.yml')
+    datasets     = []
+
+    # By default, CSVImporter only allows updating existing datasets. If the
+    # migration is adding a new dataset, add the `create_missing_datasets`
+    # keyword argument. For example:
+    #
+    #   CSVImporter.run(data_path, commits_path, create_missing_datasets: true) do |row, runner|
+    #     # ...
+    #   end
+    #
+    CSVImporter.run(data_path, commits_path) do |row, runner|
+      print "Updating #{row['geo_id']}... "
+      commits = runner.call
+
+      if commits.any?
+        datasets.push(find_dataset(commits))
+        puts 'done!'
+      else
+        puts 'nothing to change!'
+      end
+    end
+
+    sleep(1)
+    puts
+    puts "Updated #{datasets.length} datasets with the following IDs:"
+    puts "  #{datasets.map(&:id).join(',')}"
+  rescue ActiveRecord::RecordInvalid => e
+    if e.record.is_a?(Commit) && e.record.errors['dataset_edits.value'].any?
+      warn('')
+      warn('-' * 80)
+      warn('The following errors occurred while processing CSV rows:')
+      warn('')
+
+      # Grab all the errors from individual datasets to show those instead. This is typically
+      # the case when a CSV specifies a value that is not allowed to be edited.
+      e.record
+        .dataset_edits
+        .reject(&:valid?)
+        .each do |edit|
+          edit.errors.each do |field, msg|
+            warn("* #{edit.commit.dataset.geo_id}: #{edit.key}: #{field} #{msg}")
+          end
+        end
+
+      warn('-' * 80)
+    end
+
+    raise e
+  end
+
+  def self.down
+    raise ActiveRecord::IrreversibleMigration
+  end
+
+  def find_dataset(commits)
+    commits.each do |commit|
+      return commit.dataset if commit&.dataset
+    end
+  end
```

Comment:
```suggestion
def change
  new_message = "Based on the valid EPC certifications and assumption that approximately 1/3th of small residential buildings have an EPC certification. The approximated value is multiplied by 4 to align with heat technology settings of ETM (fixed value)."

  Commit.where(dataset: Dataset.find_by(geo_id: :BEGM11002))
    .joins(:dataset_edits)
    .where(dataset_edits: { key: "present_number_of_buildings" })
    .last
    .update(message: new_message)
end
```
Sorry to interfere boys, but I got an alert in my email from this PR for some reason. You can remove the `commits.yml` file and the `csv` and just do this as the migration if you only wish to update the last commit message!